modules/videoio/src/cap_android_mediandk.cpp

```diff
@@ -257,6 +260,324 @@ class AndroidMediaNdkCapture : public IVideoCapture
+                }
+
+                LOGV(
+                    "[writeBytes] partial - bufferIndex=%d, bufferSize=%u, partialSize=%u, remaining inputBufferSize=%u",
+                    (int)bufferIndex, (unsigned int)bufferSize, (unsigned int)partialSize, (unsigned int)inputBufferSize
+                );
+
+                AMediaCodec_queueInputBuffer(encoder, (size_t)bufferIndex, 0, partialSize, frameIndex * 1000000L / frameRate, flags);
+                if (NULL != inputBuffer) drainEncoder(false);
+            }
+        }
+    }
+    #endif
+
+public:
+    AndroidMediaNdkVideoWriter(const cv::String& filename, int fourcc, double fps, cv::Size frameSize, const VideoWriterParameters& params)
+        : format(NULL),
+          encoder(NULL),
+          muxer(NULL),
+          #if __ANDROID_API__ >= 26
+          surface(NULL),
+          #endif
+          frameIndex(0),
+          width(0),
+          height(0),
+          frameRate(0.),
+          videoTrackIndex(-1),
+          fd(-1) {
+        open(filename, fourcc, fps, frameSize, params);
+    }
+    virtual ~AndroidMediaNdkVideoWriter() { close(); }
+
+    virtual int getCaptureDomain() const CV_OVERRIDE { return cv::CAP_ANDROID; }
+
+    virtual void write(cv::InputArray image_ ) CV_OVERRIDE
+    {
+        if (!image_.isMat()) return;
+
+        Mat image = image_.getMat();
+        if (CV_8UC3 != image.type() || image.cols > width || image.rows > height) return;
+
+        #if __ANDROID_API__ >= 26
+        ANativeWindow_Buffer buffer;
+        if (0 != ANativeWindow_lock(surface, &buffer, NULL)) {
+            LOGE("Failed to lock the surface");
+        } else {
+            if (AHARDWAREBUFFER_FORMAT_R8G8B8A8_UNORM == buffer.format) {
+                for(int row = 0; row < image.rows; row++) {
+                    Mat rowBufferMat(1, image.cols, CV_8UC4, (int32_t*)buffer.bits + row * buffer.stride);
+                    cvtColor(image.row(row), rowBufferMat, CV_BGR2RGBA);
+                }
+            }
+
+            ANativeWindow_unlockAndPost(surface);
+        }
+        #else
+        LOGV("[write] image: %d  x %d", image.cols, image.rows);
+
+        //OpenCV don't support RGB to NV12 so we need to connvert to YV12 and then manually changed it to NV12
+        Mat imageYV12;
+        cvtColor(image, imageYV12, CV_BGR2YUV_YV12);
+
+        //convert from YV12 to NV12
+        size_t yPlaneSize = width * height;
+        size_t vPlaneSize = yPlaneSize / 4;
+
+        Mat vMat = Mat( vPlaneSize, 1, CV_8UC1, imageYV12.ptr() + yPlaneSize ).clone();
+        Mat uMat = Mat( vPlaneSize, 1, CV_8UC1, imageYV12.ptr() + yPlaneSize + vPlaneSize ).clone();
+
+        uint8_t* vMatPtr = vMat.ptr();
+        uint8_t* uMatPtr = uMat.ptr();
+        uint8_t* vuMatPtr = imageYV12.ptr() + yPlaneSize;
+        for( size_t i = 0; i < vPlaneSize; i++, vMatPtr++, uMatPtr++, vuMatPtr++ ) {
+            *vuMatPtr = *uMatPtr;
+            vuMatPtr++;
+            *vuMatPtr = *vMatPtr;
+        }
```

Comment:
`cv::merge` does it for you: https://docs.opencv.org/4.x/d2/de8/group__core__array.html#ga7d7b4d6c6ee504b30a20b1680029c7b4

Reply:
Great. I searched for something like but I did't found it.
I would be great to add RGB (and all versions) => NV12 / NV21 conversion directly in cvtColor.

Reply:
Yes, It's very rare case that's why it's not there. Please file a ticket with link to this code, when PR is merged.

Reply:
I already create an issue about.
For now I'll just use merge.